pkg/pipeline/transform/transform_network_direction.go

```diff
@@ -0,0 +1,38 @@
+package transform
+
+import (
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+)
+
+const (
+	ingress = 0
+	egress  = 1
+)
+
+func reinterpretDirection(output config.GenericMap, info *api.DirectionInfo) {
+	output[info.IfDirectionField] = output[info.FlowDirectionField]
+	var srcNode, dstNode, reporter string
+	if gen, ok := output[info.SrcHostField]; ok {
+		if str, ok := gen.(string); ok {
+			srcNode = str
+		}
+	}
+	if gen, ok := output[info.DstHostField]; ok {
+		if str, ok := gen.(string); ok {
+			dstNode = str
+		}
+	}
+	if gen, ok := output[info.ReporterIPField]; ok {
+		if str, ok := gen.(string); ok {
+			reporter = str
+		}
+	}
+	if srcNode != dstNode {
+		if srcNode == reporter {
+			output[info.FlowDirectionField] = egress
+		} else if dstNode == reporter {
+			output[info.FlowDirectionField] = ingress
+		}
+	}
+}
```

Comment:
Does this give the correct result even if one or more of these values is nil? What if all of `srcNode`, `dstNode`, and `reporter` are nil?

Reply:
good catches, thanks!
I've adding more missing fields coverage in last commit
Basically:
- When no FlowDirection set, do not set IfDirection so it's consistent 
- When no reporter set, then the whole reinterpretation cannot be done so we just set `IfDirection` (for consistency) but no change in `FlowDirection`.
- When src and/or dst aren't set they just won't match reporter so FlowDirection will be unchanged

That's also added to unit tests